shell/probe/nocturnal.sh

```diff
@@ -1,30 +1,29 @@
 #!/bin/bash
 
 PRELUDE_API=${PRELUDE_API:="https://api.preludesecurity.com"}
+PRELUDE_DIR=${PRELUDE_DIR:="."}
+PRELUDE_SLEEP=${PRELUDE_SLEEP:=14440}
 
-sys=$(uname -s)-$(uname -m)
-dos=$(echo $sys | tr '[:upper:]' '[:lower:]')
+dos=$(echo $(uname -s)-$(uname -m))
 
 while :
-do
-    temp=$(mktemp)
-    location=$(curl -sL -w %{url_effective} -o $temp -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
-    test=$(echo $location | grep -o '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | head -n 1)
-
-    if [ -z "$test" ];then
-        sleep 14400
-    else
+do  
+    exe=$PRELUDE_DIR/$(uuidgen).vst
+    location=$(curl -sL -w %{url_effective} -o $exe -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
+    
+    if [ -s $exe ];then
```

Comment:
existence of the file here doesn't mean it's a test in there

ex. this throws any error message in the file too

```
mahinakaholokula@Mahinas-MacBook-Pro my-prelude % curl -sL -w %{url_effective} -o tryme.vst -H "token:itsabadone" -H "dos:darwin-arm64" -H "dat:" "https://api.preludesecurity.com"
https://api.preludesecurity.com/%  
mahinakaholokula@Mahinas-MacBook-Pro my-prelude % cat tryme.vst 
{"Message":"User is not authorized to access this resource"}%            
```

Reply:
good point. sticking with $test is probably best. reverted.